dashboard/lib/autocomplete_helper.rb

```diff
@@ -15,6 +15,12 @@ def self.format_limit(limit)
     return [MIN_LIMIT, [limit.to_i, MAX_LIMIT].min].max
   end
 
+  def self.get_query_terms(query)
+    query.strip.split(/\s+/).map do |w|
+      w.gsub(/\W/, '').upcase.presence
```

Comment:
You're stripping word characters from query terms here but I don't see a matching operation on the other side of the query - are all the names we're matching against also limited to word characters?

My concern is that we won't match, or will at least be biased against, schools with non-word characters in their names:

- Hyphens (Jonesboro-Hodge High School)
- Periods (St. Martin's Episcopal School)
- Apostrophes (Waiʻanae High School)

Based on a little reading it looks like extended-set characters will be handled properly by the `\W` class, but probably worth having a test for. (Nānākuli High)

Reply:
Ah, I also see you just moved this functionality, didn't add it.  Still interested in your take though.

Reply:
I had not thought about this. There are 17,744 schools with a non-word, non-whitespace character in the name or city. These are mostly `-` or `'` with a few others thrown in(`,`, `&`, `@`,`/`, etc.) Poking around it seems like these are inconsistent in the data.  For example, sometimes we have "WINSTON-SALEM" and sometimes "WINSTON SALEM" and sometimes I see "CHILDREN'S" and sometimes "CHILDRENS". That means that we won't get the right results if we just leave those non-word characters as literals in the search. Instead, it seems reasonable to split on non-word character as well as on whitespace. That means that if you search for "WINSTON-SALEM" we'll look for "WINSTON" and for "SALEM" which will match both "WINSTON-SALEM" and "WINSTON SALEM" Since this is an issue even in the original search, I am changing it in both and adding a few tests.